orchagent/pfcactionhandler.h

```diff
@@ -39,15 +47,28 @@ class PfcWdActionHandler
         }
 
         static void initWdCounters(shared_ptr<Table> countersTable, const string &queueIdStr);
+        void initCounters(void);
+        void commitCounters(void);
+
+        virtual bool getHwCounters(PfcWdHwStats& counters)
+        {
+            return true;
+        };
 
     private:
         struct PfcWdQueueStats
```

Comment:
>PfcWdQueueStats [](start = 15, length = 15)

inherit PfcWdHwStats?

Reply:
I'd rather not do it. Values in every struct have different meaning and value. counters in PfcWdQueueStats represent PFC WD stats, whereas values in PfcWdHwStats are snapshots from hardware based on which first ones are calculated.